code/datums/components/food_stuff.dm

```diff
@@ -238,5 +271,317 @@
+			var/obj/item/kitchen/utensil/fork/Y = user.find_type_in_hand(/obj/item/kitchen/utensil/fork)
+			if(prob(20) && istype(Y, /obj/item/kitchen/utensil/fork/plastic))
+				Y.break_utensil(user)
+			if(istype(Y))
+				REMOVE_FLAG(remaining_flags, EATING_NEEDS_A_FORK)
+
+		if (HAS_FLAG(remaining_flags, EATING_NEEDS_A_SPOON))
+			var/obj/item/kitchen/utensil/spoon/P = user.find_type_in_hand(/obj/item/kitchen/utensil/spoon)
+			if(prob(20) && istype(P, /obj/item/kitchen/utensil/spoon/plastic))
+				P.break_utensil(user)
+			if(istype(P))
+				REMOVE_FLAG(remaining_flags, EATING_NEEDS_A_SPOON)
+
+		return remaining_flags
+
+/datum/component/consume/need_utensil/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_ITEM_CONSUMED_PRE)
+	. = ..()
+
+/// Makes the food heal you every bite
+/datum/component/consume/foodheal
+	var/obj/item/F
+	var/base_heal = 1
+
+/datum/component/consume/foodheal/Initialize(var/_base_heal)
+	..()
+	if(!istype(parent, /obj/item))
+		return COMPONENT_INCOMPATIBLE
+	src.F = parent
+	src.base_heal = _base_heal
+	RegisterSignal(parent, list(COMSIG_ITEM_CONSUMED_ALL, COMSIG_ITEM_CONSUMED_PARTIAL), .proc/eat_stuff_get_heal)
+
+/datum/component/consume/foodheal/InheritComponent(datum/component/consume/foodheal/C, i_am_original, _new_base_heal)
+	if(C?.base_heal)
+		src.base_heal = C.base_heal
+	else
+		if (isnum_safe(_new_base_heal)) // C(duplicate component) wasn't initialized, so we don't know if the raw argument _new_base_heal is actually a number
+			src.base_heal = _new_base_heal
+
+/datum/component/consume/foodheal/proc/eat_stuff_get_heal(var/obj/item/I, var/obj/item/I, var/mob/M)
+	var/healing = src.base_heal
+
+	if (ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if (H.sims)
+			H.sims.affectMotive("Hunger", base_heal * 2)
+			H.sims.affectMotive("Bladder", -base_heal * 0.2)
+
+	if (F.quality >= 5)
+		boutput(M, "<span class='notice'>That tasted amazing!</span>")
+		healing *= 2
+
+	if (F.reagents && F.reagents.has_reagent("THC"))
+		boutput(M, "<span class='notice'>Wow this tastes really good man!!</span>")
+		healing *= 2
+
+
+	if (F.quality <= 0.5)
+		boutput(M, "<span class='alert'>Ugh! That tasted horrible!</span>")
+		if (prob(20) && isliving(M))
+			var/mob/living/L = M
+			L.contract_disease(/datum/ailment/disease/food_poisoning, null, null, 1) // path, name, strain, bypass resist
+		healing = 0
+
+	M.nutrition += healing * 10
+	var/cutOff = round(M.max_health / 1.8) // 100 / 1.8 is about 55.555...6 so this should work out to be around the original value of 55 for humans and the equivalent for mobs with different max_health
+	if (ishuman(M))
+		var/mob/living/carbon/human/H = M
+		if (H.traitHolder && H.traitHolder.hasTrait("survivalist"))
+			cutOff = round(H.max_health / 10) // originally 10
+
+	if(istype(F, /obj/item/reagent_containers/food/snacks/donut))
+		if(ishuman(M) && (M.job in list("Security Officer", "Head of Security", "Detective")))
+			healing *= 2
```

Comment:
seems weird to handle here rather than on the food itself

Reply:
Moved to the food items themselves